Barley-break/js/main.js

```diff
@@ -0,0 +1,101 @@
+var game = document.getElementById("game");
+var barleyBreak = new Barleybreak(game);
+barleyBreak.addEvents();
+
+function Barleybreak(game) {
+	var elements = game.querySelectorAll(".element");
+	var reset = game.querySelector(".reset");
+	var cover = game.querySelector(".cover");
+	var congratulation = game.querySelector(".congratulation");
+	var steps = game.querySelector(".steps");
+	var sumSteps = 0;
+    var neighborCheck = function(index) {
+        if(this.verification(index + 1)){
+            this.moveCell(index + 1, index);
+        }else if(this.verification(index - 1)){
+            this.moveCell(index - 1, index);
+        }else if(this.verification(index + 4)){
+            this.moveCell(index + 4, index);
+        }else if(this.verification(index - 4)){
+            this.moveCell(index - 4, index);
+        }
+        this.checkSequence();
+    }.bind(this);
+
+	this.addEvents = function() {
+		for(var a = 0; a < elements.length; a++){
+			(function (a) {
+				elements[a].addEventListener("click", function() {
+					neighborCheck(a);
+					sumSteps++;
+				});
+			})(a);
```

Comment:
you removed unnecessary public variables and don't need binding anymore. so you don't need this inner function with passed **a** neither.